### PR TITLE
Heatmap to fc

### PR DESCRIPTION
**Pull request by ynahmany**
## PR Checklist
Please check if your PR fulfills the following requirements:

- [ ] Tests for the changes have been added (for bug fixes / features)
- [ ] Docs have been added / updated (for bug fixes / features)

## PR Type
What kind of change does this PR introduce?

```
[ ] Bugfix
[ ] Feature
[ ] Code style update (formatting, local variables)
[ X] Refactoring (no functional changes, no api changes)
[ ] Build related changes
[ ] CI related changes
[ ] Documentation content changes
[ ] Other... Please describe:
```

## What is the current behavior?

Issue Number: N/A

## What is the new behavior?
Convert Heatmap folder to be functional components.

## Does this PR introduce a breaking change?
```
[ ] Yes
[ X] No
```

## Other information
@amcdnl I need some help with `HeatmapSeries`, I am getting strange behavior and cannot locate the issue. 
would appreciate your help with that.

```
const xScale = scaleBand()
        .range([0, chartWidth])
        .domain(xDomain)
        .paddingInner(series.props.padding);
```
on the new FC, xScale range is set to a map of key to NaN.

### CODE REVIEW

File: src/Heatmap/CalendarHeatmap.tsx [hunk 1]
```diff
@@ -41,94 +40,87 @@ export interface CalendarHeatmapProps extends Omit<HeatmapProps, 'data'> {
...
-                  )}`
-                }
-              />
-            }
-          />
-        }
-      />
-    )
-  };
+  const { data: domainData, yDomain, xDomain, start } = getDataDomains();
```
Comment (amcdnl) [inline src/Heatmap/CalendarHeatmap.tsx:72]: This would be perfect for useMemo

File: src/Heatmap/CalendarHeatmap.tsx [hunk 2]
```diff
@@ -41,94 +40,87 @@ export interface CalendarHeatmapProps extends Omit<HeatmapProps, 'data'> {
...
 
-    // Get the yAxis label formatting based on view type
-    const yAxisLabelFormat = view === 'year' ? (d) => weekDays[d] : () => null;
+  // Get the yAxis label formatting based on view type
+  const yAxisLabelFormat = view === 'year' ? (d) => weekDays[d] : () => null;
 
-    // Format the xAxis label for the start + n week
-    const xAxisLabelFormat = (d) =>
-      addWeeksToDate(start, d).toLocaleString('default', { month: 'long' });
+  // Format the xAxis label for the start + n week
```
Comment (amcdnl) [inline src/Heatmap/CalendarHeatmap.tsx:80]: Maybe this could be moved outside of the FC as a pure function.

File: src/Heatmap/Heatmap.tsx [hunk 3]
```diff
@@ -150,29 +153,28 @@ export class Heatmap extends Component<HeatmapProps> {
...
-        yAxisVisible={isAxisVisible(yAxis.props)}
-        className={classNames(className)}
-      >
-        {(props) => this.renderChart(props)}
-      </ChartContainer>
-    );
-  }
-}
+  };
+  //   ,
```
Comment (amcdnl) [inline src/Heatmap/Heatmap.tsx:163]: dead code

File: src/Heatmap/Heatmap.tsx [hunk 4]
```diff
@@ -49,74 +49,77 @@ export interface HeatmapProps extends ChartProps {
...
+        .range([chartHeight, 0])
+        .paddingInner(series.props.padding);
+
+      return {
+        yScale,
+        xScale,
+        data: nestedData
+      };
+    },
+    [data, xAxis, series]
```
Comment (amcdnl) [inline src/Heatmap/Heatmap.tsx:117]: You are missing some deps

**Review by amcdnl — CHANGES_REQUESTED** [review]

**Comment by ynahmany** [comment]
@amcdnl fixed all open issues.

**Review by amcdnl — APPROVED** [review]
LGTM - AWESOME STUFF!

**Pull request merged**